### DISCUSSION TITLE
esp32-c3-devkitm-1u BMP280 Crash Dump

### DISCUSSION THREAD

**Discussion by jimcondit** [UncleRus/esp-idf-lib#264] [Q&A] [answered]
First time using the UncleRus library.  While trying to use the example code for the BMP280, and after modifying the SLA and SLC pins (respectively GPIO5, GPIO6), all compiles nicely but throws a crash dump during execution.  (Shown below.)  If I believe the dump, it appears to be crashing at the bmp280_init call.  Also, it appears to be looking for the device at 0x76.  But, when using espressif 12c_tools code, it finds the device at 0x77, which appears to be correct from other resource.  Did I miss some configuration parameter?

Thanks for your help!

Jim

I (940) cpu_start: Starting scheduler.
E (945) i2cdev: Could not read from device [0x76 at 0]: -1
E (945) bmp280: Sensor not found
ESP_ERROR_CHECK failed: esp_err_t 0xffffffff (ESP_FAIL) at 0x40384c7b
0x40384c7b: _esp_error_check_failed at /home/condit/esp/esp-idf/components/esp_system/esp_err.c:42

file: "../main/main.c" line 28
func: bmp280_test
expression: bmp280_init(&dev, &params)

abort() was called at PC 0x40384c7f on core 0
0x40384c7f: _esp_error_check_failed at /home/condit/esp/esp-idf/components/esp_system/esp_err.c:43

Stack dump detected
Core  0 register dump:
MEPC    : 0x40380664  RA      : 0x40384c8a  SP      : 0x3fc929d0  GP      : 0x3fc8c000  
0x40380664: panic_abort at /home/condit/esp/esp-idf/components/esp_system/panic.c:402

0x40384c8a: __ubsan_include at /home/condit/esp/esp-idf/components/esp_system/ubsan.c:294

TP      : 0x3fc8aae0  T0      : 0x37363534  T1      : 0x7271706f  T2      : 0x33323130  
S0/FP   : 0x00000004  S1      : 0x3fc92a34  A0      : 0x3fc929fc  A1      : 0x3fc92a32  
A2      : 0x00000000  A3      : 0x3fc92a29  A4      : 0x00000001  A5      : 0x3fc8e000  
A6      : 0x7a797877  A7      : 0x76757473  S2      : 0x00000000  S3      : 0x00000000  
S4      : 0x00000000  S5      : 0x00000000  S6      : 0x00000000  S7      : 0x00000000  
S8      : 0x00000000  S9      : 0x00000000  S10     : 0x00000000  S11     : 0x00000000  
T3      : 0x6e6d6c6b  T4      : 0x6a696867  T5      : 0x66656463  T6      : 0x62613938  
MSTATUS : 0x00001881  MTVEC   : 0x40380001  MCAUSE  : 0x00000007  MTVAL   : 0x00000000  
0x40380001: _vector_table at ??:?

MHARTID : 0x00000000  

Failed to run gdb_panic_server.py script: Command '['riscv32-esp-elf-gdb', '--batch', '-n', '/home/condit/esp/bmp280/build/example-bmp280.elf', '-ex', 'target remote | "/home/condit/.espressif/python_env/idf5.0_py3.8_env/bin/python" "/home/condit/esp/esp-idf/tools/idf_monitor_base/../gdb_panic_server.py" --target esp32c3 "/tmp/tmp_8lroo2r"', '-ex', 'bt']' returned non-zero exit status 127.
b'riscv32-esp-elf-gdb: error while loading shared libraries: libpython2.7.so.1.0: cannot open shared object file: No such file or directory\n'

Core  0 register dump:
MEPC    : 0x40380664  RA      : 0x40384c8a  SP      : 0x3fc929d0  GP      : 0x3fc8c000  
TP      : 0x3fc8aae0  T0      : 0x37363534  T1      : 0x7271706f  T2      : 0x33323130  
S0/FP   : 0x00000004  S1      : 0x3fc92a34  A0      : 0x3fc929fc  A1      : 0x3fc92a32  
A2      : 0x00000000  A3      : 0x3fc92a29  A4      : 0x00000001  A5      : 0x3fc8e000  
A6      : 0x7a797877  A7      : 0x76757473  S2      : 0x00000000  S3      : 0x00000000  
S4      : 0x00000000  S5      : 0x00000000  S6      : 0x00000000  S7      : 0x00000000  
S8      : 0x00000000  S9      : 0x00000000  S10     : 0x00000000  S11     : 0x00000000  
T3      : 0x6e6d6c6b  T4      : 0x6a696867  T5      : 0x66656463  T6      : 0x62613938  
MSTATUS : 0x00001881  MTVEC   : 0x40380001  MCAUSE  : 0x00000007  MTVAL   : 0x00000000  
MHARTID : 0x00000000  

Stack memory:
3fc929d0: 0x3c022e88 0x40384c7e 0x3fc92a30 0x4038a2c8 0x3fc92ac8 0x3fc92a74 0x3fc92a98 0x3fc8c688
3fc929f0: 0x3fc92a34 0x3fc8c6a4 0x3fc92a30 0x726f6261 0x20292874 0x20736177 0x6c6c6163 0x61206465
3fc92a10: 0x43502074 0x34783020 0x34383330 0x20663763 0x63206e6f 0x2065726f 0x00000030 0x00000000
3fc92a30: 0xa5a50030 0x38333034 0x66376334 0x00000000 0x00000000 0x00000000 0x00000000 0x40384c82
3fc92a50: 0x00000000 0x00000000 0x00000000 0x42005bc4 0xa5a5a5a5 0xa5a5a5a5 0xa5a5a5a5 0xa5a5a5a5
3fc92a70: 0x42005b44 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000
3fc92a90: 0x00000000 0x00000000 0x00000000 0x00000000 0x00000005 0x00000006 0x00000000 0x000f4240
3fc92ab0: 0x00000000 0x00000000 0x00000076 0x3fc92cf8 0x00000000 0x00000000 0x00000003 0x00000000
3fc92ad0: 0x00000003 0x00000003 0x00000003 0x00000003 0x00000000 0x00000000 0x00000000 0x403874bc
3fc92af0: 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0xa5a5a5a5 0xa5a5a5a5 0xa5a5a5a5
3fc92b10: 0x27146414 0x01010107 0x00000001 0x00000000 0xffffffff 0x7fefffff 0x00000000 0x3fc00000
3fc92b30: 0x00000000 0x40300000 0x00000000 0x3fe00000 0x00000000 0x3ff80000 0x636f4361 0x3fd287a7
3fc92b50: 0x8b60c8b3 0x3fc68a28 0x509f79fb 0x3fd34413 0x00000000 0x3ff00000 0x00000000 0x40240000
3fc92b70: 0x00000000 0x401c0000 0x00000000 0x40140000 0x00000000 0x43500000 0x3fc8b800 0x00001480
3fc92b90: 0xa5a5a5a5 0xa5a5a5a5 0xa5a5a5a5 0x00000154 0x3fc928e0 0x00000064 0x3fc8d51c 0x3fc8d51c
3fc92bb0: 0x3fc92ba0 0x3fc8d514 0x00000014 0x3fc92e44 0x3fc92e44 0x3fc92ba0 0x00000000 0x00000005
3fc92bd0: 0x3fc9139c 0x32706d62 0x745f3038 0x00747365 0x00eb5ed9 0x00000000 0x3fc92b90 0x00000005
3fc92bf0: 0x00000000 0x00000000 0x00000000 0x00000000 0x3fc8e4bc 0x3fc8e524 0x3fc8e58c 0x00000000
3fc92c10: 0x00000000 0x00000001 0x00000000 0x00000000 0x00000000 0x4200c278 0x00000000 0x00000000
3fc92c30: 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000
3fc92c50: 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000
3fc92c70: 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000
3fc92c90: 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000
3fc92cb0: 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000
3fc92cd0: 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000
3fc92cf0: 0x3fc90000 0x00000054 0x00000000 0x3fc92cf8 0x00000000 0x00000000 0x00000000 0x3fc92d10
3fc92d10: 0xffffffff 0x3fc92d10 0x3fc92d10 0x00000000 0x3fc92d24 0xffffffff 0x3fc92d24 0x3fc92d24
3fc92d30: 0x00000001 0x00000001 0x00000000 0x0100ffff 0x00000000 0xe71e8735 0x3fc92cf0 0x0000006c
3fc92d50: 0x00000000 0x00000001 0x3fc92e74 0x00000000 0x00000004 0x00000000 0x00000000 0x00000000
3fc92d70: 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x3fc92edc 0x3fc92f3c
3fc92d90: 0x3fc92eac 0x00000000 0x00000000 0x3fc92e18 0x3fc92dc0 0x00000000 0x00000000 0x00000000
3fc92db0: 0x00000000 0x00000000 0x00000000 0x00000054 0x00000000 0x3fc92dc0 0x00000000 0x00000000

ELF file SHA256: ebe48ad7b0505c0e
ELF file SHA256: ebe48ad7b0505c0e

Rebooting...
ESP-ROM:esp32c3-api1-20210207
Build:Feb  7 2021
rst:0x3 (RTC_SW_SYS_RST),boot:0xc (SPI_FAST_FLASH_BOOT)
Saved PC:0x403804c6
0x403804c6: esp_restart_noos_dig at /home/condit/esp/esp-idf/components/esp_system/esp_system.c:46 (discriminator 1)

**Comment by UncleRus** [comment, marked as answer]
Why it's crashing? It's crashing due to the call `ESP_ERROR_CHECK()` macro on the line 28 of `main.c` in example:
```C
ESP_ERROR_CHECK(bmp280_init(&dev, &params));
```
`bmp280_init()` tries to communicate with the device by incorrect I2C address, returns `ESP_FAIL` and `ESP_ERROR_CHECK()` reboots the chip.

Here's what you need to do:

1. Correct the I2C address. In line 27 of `main.c` in example replace `BMP280_I2C_ADDRESS_0` with `BMP280_I2C_ADDRESS_1`:
```C
ESP_ERROR_CHECK(bmp280_init_desc(&dev, BMP280_I2C_ADDRESS_1, 0, SDA_GPIO, SCL_GPIO));
```
2. Replace GPIO numbers with yours in lines 12 and 13 of `main.c`
```C
#define SDA_GPIO 5
#define SCL_GPIO 6
```
3. Compile and flash the example.

**Reply by jimcondit** [reply]
Still crashes.  However, I found a bmp180 and set the GPIO ports and it worked perfectly!  Thanks so very much for your answer!
